source/utils.f90

```diff
@@ -70,4 +72,346 @@ real(kind=fPrec) function lininterp(x,xvals,yvals,datalen)
 
   end function lininterp
 
+  ! ================================================================================================ !
+  !  A.Mereghetti, CERN, BE-ABP-HSS
+  !  Last modified: 15-04-2019
+  !  Make some sanity checks on arrays to be interpolated
+  ! ================================================================================================ !
+  logical function checkArray(xvals,datalen,lprint)
+
+    use crcoall
+    implicit none
+
+    integer,           intent(in) :: datalen
+    real(kind=fPrec),  intent(in) :: xvals(1:datalen)
+    logical, optional, intent(in) :: lprint
+
+    integer ii
+    logical llprint
+
+    llprint = .true.
+    if(present(lprint)) llprint = lprint
+    checkArray=.true.
+
+    if(datalen <= 0) then
+      checkArray=.false.
+      if (llprint) write(lout,"(a)") "UTILS> ERROR checkArray: datalen was 0!"
+    end if
+    do ii=1, datalen-1
+      if(xvals(ii) >= xvals(ii+1)) then
+        checkArray=.false.
+        if (llprint) write(lout,"(a)") "UTILS> ERROR checkArray: xvals should be in increasing order"
+      end if
```

Comment:
These should now be printed to `lerr`, not `lout`. Also, do these errors cause an exit somewhere based on the checkArray flag?

Reply:
done.
For the moment, `checkArray` is not used, but I will plug it in into the main code soon